docs/beyond_basics/101-181-metalad.rst

```diff
@@ -4,10 +4,363 @@ Metadata-Management with MetaLad
+Regardless of whether it is a distribution or a retrieval process, though, an export with MetaLad will only concern the *metadata*, and never the primary data.
+
+Download Metadata from a remote repository
+""""""""""""""""""""""""""""""""""""""""""
+
+Let's start by creating a place where someone else's metadata could live.
+
+.. runrecord:: _examples/DL-101-181-110
+   :language: console
+   :workdir: beyond_basics/meta
+
+   $ datalad create metadata-assimilation
+   $ cd metadata-assimilation
+
+Because MetaLad stores metadata in :term:`Git`'s object store, we use Git to directly fetch metadata from a remote repository, such as this demo on :term:`GitHub`: ``https://github.com/christian-monch/metadata-test.git``.
+Because metadata added by MetaLad is not transported automatically but needs to be specifically requested, the command to retrieve it looks unfamiliar to non-Git-users: It identifies the precise location of the :term:`ref` that contains the metadata.
+
+.. find-out-more:: Exactly where is metadata stored?
+   :name: fom-metadataobjecttree
+
+   MetaLad employs an internal metadata model that makes the following properties possible:
+
+   * Metadata has a version encoded, but isn't itself version controlled
+   * Metadata should not be transported if not explicitly requested
+   * It should be possible to only retrieve parts of the overall metadata tree, e.g. certain sub-nodes
+
+   TODO this needs much more
+
+.. runrecord:: _examples/DL-101-181-111
+   :language: console
+   :workdir: beyond_basics/meta/metadata-assimilation
+
+   $ git fetch \
+      "https://github.com/christian-monch/metadata-test.git" \
+      "refs/datalad/*:refs/datalad/*"
+
+The metadata is now locally available in the Git repository ``metadata-repo``.
+You can verify this by issuing the command ``datalad meta-dump -r``, which will list all metadata in the repository.
+Can you guess what type of metadata it contains [#f5]_ ?
+
+.. runrecord:: _examples/DL-101-181-112
+   :language: console
+   :workdir: beyond_basics/meta/metadata-assimilation
+
+   $ datalad meta-dump -r
+
+Publish metadata to a Git-Repository
+""""""""""""""""""""""""""""""""""""
+
+You can also push your metadata to a remote :term:`sibling` (if you have write :term:`permissions`).
+This, too, uses a Git command to push only specific :term:`ref`\s.
+Let's assume you are in the directory that contains the git repository with your metadata, then you can push your metadata to a remote git repository ``<your repository>``::
+
+    $ git push "<your repository>" "refs/datalad/*:refs/datalad/*"
+
+You will notice that no primary data is stored in the repository ``metadata-destination``. That allows you to publish metadata without publishing the primary data at the same time.
+
+
+Querying metadata
+^^^^^^^^^^^^^^^^^
+
+TODO: something more about meta-dump and concrete usage example with, e.g., ``jq``
```

Comment:
Does anyone has an example in their shell history, by any chance? :)

Reply:
This would create a list of all ``familyName`` values in the ``person`` lists that were extracted from ``.studyminimeta.yaml``-files by ``metalad_studyminimeta``-extractor:

```shell
> datalad meta-dump -r|jq '.extracted_metadata["@graph"][3]["@list"][].familyName'|sort|uniq
```

Reply:
Something like this? (also studyforrest dataset)

Output array of all subdatasets:

```
$ datalad meta-extract -d . metalad_core > metadata_record_metalad_core.json
$ datalad meta-add -d . metadata_record_metalad_core.json
$ datalad meta-dump . | jq '.extracted_metadata["@graph"] | .[] | select(.["@type"] == "Dataset") | [.hasPart[]? | {"dataset_id": (.identifier | sub("^datalad:"; "")), "dataset_version": (.["@id"] | sub("^datalad:"; "")), "dataset_path": .name}]'

[
  {
    "dataset_id": "d5dd3da0-a631-4c0c-a4a9-de55dfc4620f",
    "dataset_version": "276ceffd8c11db1b99b30b4ae6965aac296fba02",
    "dataset_path": "artifact/3T_movie_eyetracking"
  },
  {
    "dataset_id": "ad9b6c66-4413-4b4f-b6da-b7f25d0d6397",
    "dataset_version": "787741801ad6ff9e4d0aea7a1403bc483f9e1a74",
    "dataset_path": "artifact/3T_structural_mri"
  },
  {
    "dataset_id": "6075c0fa-ab72-4bab-9888-3b597f0e63b1",
    "dataset_version": "64cf42ad00c707bb6533b5944041582c6bfdd384",
    "dataset_path": "artifact/3T_visuallocalizer"
  },
  {
    "dataset_id": "d4759300-5563-467d-be5f-e5b164fb3060",
    "dataset_version": "7b6ec16910eb6a77cd0cc3d2945e7fb292fb9672",
    "dataset_path": "artifact/7T_audiomovie"
  },
  {
    "dataset_id": "c08af312-e05b-43b3-b499-db0d2ad46bf6",
    "dataset_version": "7b58b38af7167d576489cc58e9ec4b3eee0822fc",
    "dataset_path": "artifact/7T_musicperception"
  },
  {
    "dataset_id": "da15d84c-9c8b-11e9-a3fb-f0d5bf7b5561",
    "dataset_version": "f168e8e2b8bef21f373e8a5385e528130ca6339d",
    "dataset_path": "artifact/media"
  },
  {
    "dataset_id": "126cd950-377c-4600-a921-045cf408bd9f",
    "dataset_version": "1b7cb46a97e75aeccab28ef7b61693f4f02a0f8d",
    "dataset_path": "artifact/movie_eyetracking"
  },
  {
    "dataset_id": "0f66b1ba-e9a9-46fd-b9d9-2e64fe94d307",
    "dataset_version": "c74b66cf37c0d4ed8914296c6d7792b2d25696aa",
    "dataset_path": "code/conversion_qa"
  },
  {
    "dataset_id": "7fcd8812-d0fe-11e7-8db2-a0369f7c647e",
    "dataset_version": "2ccaa115543c21e6658950d1cb8cc3038f14272f",
    "dataset_path": "derivative/aggregate_fmri_timeseries"
  },
  {
    "dataset_id": "c8ec2919-493b-4af5-9271-cbe9ebd08c43",
    "dataset_version": "74cd7ec0538448b05fb4d5f91119b279c5e9ab04",
    "dataset_path": "derivative/aligned_mri"
  },
  {
    "dataset_id": "3304e775-5f5f-435a-b68e-d98c9f5fb72a",
    "dataset_version": "aaac44e047d375cd8f791b1b6fe2b739f02c83b2",
    "dataset_path": "derivative/cortical_surfaces_freesurfer"
  },
  {
    "dataset_id": "ceb007ac-ef05-4392-98d2-35c02a774a21",
    "dataset_version": "688d8d8558fe847f4c1b19aed579745bcd6c7744",
    "dataset_path": "derivative/image_space_transformations"
  },
  {
    "dataset_id": "2d05f277-94b0-470b-8e11-4e56691d5b89",
    "dataset_version": "78e04a00fedbe3e055f86c2f9127aa48e1133d55",
    "dataset_path": "derivative/retinotopic_maps"
  },
  {
    "dataset_id": "92e65958-4a5a-4c34-a4f4-ee070f7a123b",
    "dataset_version": "203aa983534fc2b823ff4777a85f4f80d7a68656",
    "dataset_path": "derivative/visual_areas"
  },
  {
    "dataset_id": "5b1081d6-84d7-11e8-b00a-a0369fb55db0",
    "dataset_version": "b623351b43eb2715331ac59ad4cf41682e84ff7d",
    "dataset_path": "original/3T_multiresolution_fmri"
  },
  {
    "dataset_id": "1882e2e6-fbbf-4ade-a65f-3a1615235f51",
    "dataset_version": "e5d2f8368fc5f6717d8ef131041c6d943298d0c7",
    "dataset_path": "original/3T_structural_mri"
  },
  {
    "dataset_id": "3a8648b3-7df8-413f-8efb-4d39040ac174",
    "dataset_version": "10e23aafa8271f742e9022a3522d9a88d7fe30cf",
    "dataset_path": "original/7T_multiresolution_fmri"
  },
  {
    "dataset_id": "5eaff716-54eb-11e8-803d-a0369f7c647e",
    "dataset_version": "72f835ada046bd0479009ea0ff933b30a95b0076",
    "dataset_path": "original/phase2"
  },
  {
    "dataset_id": "4c536c4a-ec61-11e6-9440-00b56d060aa7",
    "dataset_version": "298e2a884d0598dd89753e9b5576bd93d19f335a",
    "dataset_path": "stimulus/computational_annotations"
  },
  {
    "dataset_id": "45b9ab26-07fc-11e8-8c71-f0d5bf7b5561",
    "dataset_version": "29dcce2b9477537b433996ebc342c531139e1d87",
    "dataset_path": "stimulus/curated_annotations"
  }
]
```

Reply:
Ooh I only saw your reply now @christian-monch. I think yours is a bit more user friendly :)

Reply:
I think it is good example. Just wanted to mention that the first two lines can be combined to:

```shell
> datalad meta-extract -d . metalad_core  | datalad meta-add -d . --json-lines -
``` 

And since ``-d .`` is equivalent to the default dataset location, the following would be identical:
```shell
> datalad meta-extract metalad_core  | datalad meta-add --json-lines -
```

Reply:
Thanks a lot! I think the family name example would work really well. I get a few errors/warnings by jq in @christian-monch's example - any idea what might be wrong?

```
datalad meta-dump -r | jq '.extracted_metadata["@graph"][3]["@list"][].familyName' | sort | uniq
jq: error (at <stdin>:16): Cannot iterate over null (null)
jq: error (at <stdin>:22): Cannot iterate over null (null)
jq: error (at <stdin>:23): Cannot iterate over null (null)
jq: error (at <stdin>:24): Cannot iterate over null (null)
```

Reply:
This is yet unresolved, I'm merely hiding it in the output. If some has advice, let me know